### DISCUSSION TITLE
Saving the Initialized SDK to Enviroment variables

### DISCUSSION THREAD

**Discussion by AniTexs** [Mark5900/Capa.PowerShell.Module#371] [Ideas]
Is there a reason why the Initialized sdk is not saved to enviroment so all modules can pull that instead of it needing to be passed to all cmdlets? :)

**Comment by Mark5900** [comment]
The best reason I can think about is for the readability of ones script if you have multiple environments.

If I have done what you suggested could there be scenarios where you had to run Initialize-CapaSDK multiple times to switch between the environments? Then it makes more sense to define if a cmdlet is to be run against ones production or development environment.

Example
```powershell
$oCMSProd = Initialize-CapaSDK -Server $Settings.CapaServer -Database $Settings.Database -InstanceManagementPoint $Settings.InstanceManagementPointProd
$oCMSDev = Initialize-CapaSDK -Server $Settings.CapaServer -Database $Settings.Database -InstanceManagementPoint $Settings.InstanceManagementPointDev

# Remove-OldUnits is a custom function in my script that deletes old use units
Remove-OldUnits -CapaSDK $oCMSDev -Settings $Settings
Remove-OldUnits -CapaSDK $oCMSProd -Settings $Settings
```

**Reply by AniTexs** [reply]
Understand this segmentation, it could be two arrays (In case of multiple Prod and Dev instances) so you have a mandatory enviroment paramater which is called -Enviroment which is a validate set of Development or Production, this helps the function save to the correct array. Then a function that is named something like Get-CapaSDK and that will return both arrays if both have SDK's in them, otherwise it will always return only single latest SDK initialized, is this something that would work? :)

**Reply by AniTexs** [reply]
Another option could also to save the latest initialized SDK to an ENV variables and all functions will use that if no CapaSDK is specificed?

**Reply by Mark5900** [reply]
Tænker du er dansker? Så vi kan også tage den på dansk ☺️

I think the way it is done now is still the the preferred way as CapaInstaller can be set up very complex and with more than a dev and prod environment.

So the modules should be as easy to use as possible, and I think your suggestions will make it more complex for one who is new to PowerShell to understand.

**Reply by AniTexs** [reply]
Roger :)
Så vil jeg ikke sætte mine forhåbninger op omkring det :)

**Reply by Mark5900** [reply]
Lyder godt 😊
Ellers siger du bare til hvis der er noget, min kollega og jeg er i hvert fald blevet glad for modulet og vi prøver at lave alt i PowerShell frem for VBscript.

Vil anbefale at du tager med til CUG næste gang, som planen er pt. skal jeg snakke om PowerShell i 1 times tid.

**Reply by AniTexs** [reply]
Har ikke været med de sidste par gange men det skal jeg huske og se om jeg kan sætte tid at til at komme med :)
